src/Http/GenericRouter.php

```diff
@@ -82,6 +77,25 @@ public function dispatch(Request $request): Response
         return $this->createResponse($outputFromController);
     }
 
+    private function getCallable(
+        Route $route,
+        array $routeParams
+    ): Closure {
+        $callable = fn (Request $request) => $this->container->call(
+            $this->container->get($route->handler->getDeclaringClass()->getName()),
+            $route->handler->getName(),
+            ...$routeParams,
+        );
+
+        $middlewareStack = $route->middleware;
+
+        while ($middlewareClass = array_pop($middlewareStack)) {
```

Comment:
Is there a reason we are using `while` here? My biggest concern is that it looks like we are prioritizing the middleware registered _last_, but we might actually want to be prioritizing middleware registered _first_?

`foreach` is generally faster too, though the speed difference is probably so minimal it is not worth mentioning.

Reply:
Well you _need_ to apply the middleware in reverse order, so either your first reverse the array and then loop over it, or you use while + array_pop. I doubt the the reverse + foreach approach would be noticeably faster.